src/addons/meta/cursor.c

```diff
@@ -129,7 +130,12 @@ ecs_meta_type_op_t* flecs_meta_cursor_get_ptr(
     const EcsOpaque *opaque = scope->opaque;
 
     if (scope->vector) {
-        ecs_vec_set_min_count(NULL, scope->vector, size, scope->elem_cur + 1);
+        /* Prepare one element in advance to return that can be set later. 
+         * ecs_meta_next checks vector capacity and wants one extra 
+         * thus adding 2 extra in ecs_vec_set_size */
```

Comment:
This would mean a vector gets modified when a user calls `ecs_meta_get_ptr` which should be a readonly operation.